### PR TITLE
Validate: Unique variable names

### PR DESCRIPTION
**Pull request by jjergus**

### CODE REVIEW

**Comment by dschafer** [comment]
Looks good to me; I might change the field name to a real field in the sample schema to make it more clear that the valid queries are truly valid (and that the invalid queries are only invalid because of this validation rule).

**Comment by jjergus** [comment]
Sure. However, the queries are also invalid because the variables are unused. Should we address that as well? Other validation tests don't, as far as I can tell.

**Comment by leebyron** [comment]
You need to also add this to the default list of validation rules in https://github.com/graphql/graphql-js/blob/master/src/validation/specifiedRules.js

**Comment by leebyron** [comment]
Otherwise this looks great. @jjergus do you want to also draft a PR to the spec to include this?

File: src/validation/__tests__/UniqueVariableNames.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    expectPassesRule(UniqueVariableNames, `
+      query A($x: Int, $y: String) { __typename }
+      query B($x: String, $y: Int) { __typename }
+    `);
+  });
+
+  it('duplicate variable names', () => {
+    expectFailsRule(UniqueVariableNames, `
+      query A($x: Int, $x: Int, $x: String) { __typename }
+      query B($x: String, $x: Int) { __typename }
```
Comment (leebyron) [inline src/validation/__tests__/UniqueVariableNames.js]: perhaps also include a test like `query C($x: Int, $x: Int) { __typename }` just to illustrate that *only* duplicating a variable name is still not ok

**Comment by jjergus** [comment]
> do you want to also draft a PR to the spec to include this?

https://github.com/facebook/graphql/pull/145

File: src/validation/__tests__/UniqueVariableNames.js (hunk 1 above)
Comment (jjergus) [inline src/validation/__tests__/UniqueVariableNames.js]: Done.

**Comment by leebyron** [comment]
Excellent

**Pull request merged**